### PR TITLE
Adding Ski Passes on Purchase Completed

### PR DESCRIPTION
**Pull request by dpaquette**
Looking for feedback on the second commit here.
Where should the notifications and handlers live? Did I wire up MediatR correctly?

### CODE REVIEW

File: src/AlpineSkiHouse.Web/Events/PassPurchase.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using MediatR;
+
+namespace AlpineSkiHouse.Events
+{
+    public class PassPurchase
+    {
+        public int PassTypeId { get; set; }
+        public int CardId { get; set; }
```
Comment (MisterJames) [inline src/AlpineSkiHouse.Web/Events/PassPurchase.cs]: I think if we were in the real world of things we might want to put the complete details in, either here or in the Purchase completed events, to capture things like the price paid for the pass, discount code used, etc. Do you think we could either a) ignore it completely or b) make a note in the source comments or in the book? Your chapter is already stoopid-hyooge, so I don't want to push too much in that would cloud the book focus, but at the same time we might want to note it (could even be a two-sentence side bar).

File: src/AlpineSkiHouse.Web/Startup.cs [hunk 2]
```diff
@@ -13,6 +13,8 @@
 using AlpineSkiHouse.Models;
 using AlpineSkiHouse.Services;
 using Microsoft.AspNetCore.Mvc;
+using MediatR;
+using Scrutor;
```
Comment (MisterJames) [inline src/AlpineSkiHouse.Web/Startup.cs]: What is "Scrutor"?

Comment (dpaquette) [inline src/AlpineSkiHouse.Web/Startup.cs]: It’s a package that does assembly scanning for aspnetcore di. I got it from the mediatr sample

File: src/AlpineSkiHouse.Web/Events/PassPurchase.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿using MediatR;
+
+namespace AlpineSkiHouse.Events
+{
+    public class PassPurchase
```
Comment (stimms) [inline src/AlpineSkiHouse.Web/Events/PassPurchase.cs]: Typically events are named in the past tense. `PassPurchased`

File: src/AlpineSkiHouse.Web/Events/PassPurchase.cs (hunk 1 above)
Comment (stimms) [inline src/AlpineSkiHouse.Web/Events/PassPurchase.cs]: For events internal to the system I agree with making them "fat" events.

Comment (dpaquette) [inline src/AlpineSkiHouse.Web/Events/PassPurchase.cs]: Sounds good. I'll add more details

File: src/AlpineSkiHouse.Web/Events/PassPurchase.cs (hunk 3 above)
Comment (dpaquette) [inline src/AlpineSkiHouse.Web/Events/PassPurchase.cs]: Maybe I approached this the wrong way. There is a PurchaseCompleted event which contains a number of PassPurchases. Would the individual line items also be named as events? Maybe instead I should be raising a number of individual PassPurchased events...

File: src/AlpineSkiHouse.Web/Handlers/AddSkiPassOnPurchaseCompleted.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+            foreach (var newPass in newPasses)
+            {
+                var passAddedEvent = new PassAdded
+                {
+                    PassId = newPass.Id,
+                    PassTypeId = newPass.PassTypeId,
+                    CardId = newPass.CardId,
+                    CreatedOn = newPass.CreatedOn
+                };
```
Comment (dpaquette) [inline src/AlpineSkiHouse.Web/Handlers/AddSkiPassOnPurchaseCompleted.cs:46]: Going to add automapper in later. Will mention it in the chapter

**Comment by dpaquette** [comment]
This PR look OK?

File: src/AlpineSkiHouse.Web/Events/PassPurchased.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿namespace AlpineSkiHouse.Events
+{
+    public class PassPurchased
+    {
+        public int PassTypeId { get; set; }
+        public int CardId { get; set; }
+
+        public decimal PricePaid { get; set; }
+
+        public string DiscountCode { get; set; }
```
Comment (stimms) [inline src/AlpineSkiHouse.Web/Events/PassPurchased.cs:10]: I always like to see dates on events but it is less important for internal messages.

File: src/AlpineSkiHouse.Web/Handlers/AddSkiPassOnPurchaseCompleted.cs (hunk 4 above)
Comment (stimms) [inline src/AlpineSkiHouse.Web/Handlers/AddSkiPassOnPurchaseCompleted.cs:46]: Cool, cool.

File: src/AlpineSkiHouse.Web/project.json [hunk 6]
```diff
@@ -40,7 +40,9 @@
             "type": "build"
         },
         "Microsoft.AspNetCore.Authentication.Facebook": "1.0.0",
-        "Microsoft.AspNetCore.Authentication.Twitter": "1.0.0"
+        "Microsoft.AspNetCore.Authentication.Twitter": "1.0.0",
+        "MediatR": "2.1.0",
+        "MediatR.Extensions.Microsoft.DependencyInjection": "1.0.0"
```
Comment (stimms) [inline src/AlpineSkiHouse.Web/project.json:45]: That's cool, I didn't know there was a package for this.

Comment (dpaquette) [inline src/AlpineSkiHouse.Web/project.json:45]: It is very new

**Comment by MisterJames** [comment]
:shipit:

**Pull request merged**